File: solutions/typescript/luhn/1/luhn.ts
export function valid(digitString: string): unknown {
  let validLuhn: boolean = false
  let luhnString: string = digitString.replace(/\s/g, '')
  let sum: number=0
  let luhnLen: number = luhnString.length
  if (isNaN(parseInt(luhnString))) {
    validLuhn = false
  }
  else {
    if (luhnLen == 1){
      validLuhn = false
    }
    else {
      for (let i = 0; i < luhnLen; i++) {
        let digit = Number(luhnString.charAt(luhnLen-i-1))
        if (i % 2 !=0){
          digit=digit*2
          if (digit > 9){
            digit=digit-9
          }
        }
        sum=sum+digit
      }
      if (sum % 10 == 0){
        validLuhn=true
      }
    }
  }
  return validLuhn
}

File: solutions/python/luhn/1/luhn.py
class Luhn:
    def __init__(self, card_num):
        self.card_num=card_num

    def valid(self):
        valid_luhn = False
        self.card_num=self.card_num.replace(" ","")
        if len(self.card_num)>1 and self.card_num.isdigit():
            acc=0
            for i in range(len(self.card_num)):
                element=int(self.card_num[-i-1])
                if i%2 != 0:
                    element=2 * element
                    if element>9:
                        element=element-9
                acc=acc + element
            if acc % 10 == 0:
                valid_luhn=True
        return valid_luhn
